doc/source/cluster/kubernetes/examples/argocd.md

```diff
@@ -0,0 +1,461 @@
+
+## Prerequisites
+
+Before proceeding with this guide, ensure you have the following:
+
+* A Kubernetes cluster with appropriate resources for running Ray workloads.
+* `kubectl` configured to access your Kubernetes cluster.
+* (Optional)[ArgoCD installed](https://argo-cd.readthedocs.io/en/stable/getting_started/) on your Kubernetes cluster.
+* (Optional)[ArgoCD CLI](https://argo-cd.readthedocs.io/en/stable/cli_installation/) installed on your local machine (recommended for easier application management).
+* (Optional)Access to the ArgoCD UI or API server.
+
+## Step 1: Deploy KubeRay Operator CRDs
+
+First, deploy the Custom Resource Definitions (CRDs) required by the KubeRay operator. Create a file named `ray-operator-crds.yaml` with the following content:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: ray-operator-crds
+  namespace: argocd
+spec:
+  project: default
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: ray-cluster
+  source:
+    repoURL: https://github.com/ray-project/kuberay
+    targetRevision: v1.4.0  # update this as necessary
+    path: helm-chart/kuberay-operator/crds
+  syncPolicy:
+    automated:
+      prune: true
+      selfHeal: true
+    syncOptions:
+      - CreateNamespace=true
+      - Replace=true
+```
+
+Apply the ArgoCD Application:
+
+```sh
+kubectl apply -f ray-operator-crds.yaml
+```
+
+Wait for the CRDs Application to sync and become healthy. You can check the status using:
+
+```sh
+kubectl get application ray-operator-crds -n argocd
+```
+
+Alternatively, if you have the ArgoCD CLI installed, you can wait for the application:
+
+```sh
+argocd app wait ray-operator-crds
+```
+
+## Step 2: Deploy the KubeRay Operator
+
+After the CRDs are installed, deploy the KubeRay operator itself. Create a file named `ray-operator.yaml` with the following content:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: ray-operator
+  namespace: argocd
+spec:
+  project: default
+  source:
+    repoURL: https://github.com/ray-project/kuberay
+    targetRevision: v1.4.0  # update this as necessary
+    path: helm-chart/kuberay-operator
+    helm:
+      skipCrds: true   # CRDs are already installed in Step 1
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: ray-cluster
+  syncPolicy:
+    automated:
+      prune: true
+      selfHeal: true
+    syncOptions:
+      - CreateNamespace=true
+```
+
+Note the `skipCrds: true` setting in the Helm configuration. This is required because the CRDs were installed separately in Step 1.
+
+Apply the ArgoCD Application:
+
+```sh
+kubectl apply -f ray-operator.yaml
+```
+
+Wait for the operator Application to sync and become healthy. You can check the status using:
+
+```sh
+kubectl get application ray-operator -n argocd
```

Comment:
It would be better to provide some example output of the command to verify.
```
NAME           SYNC STATUS   HEALTH STATUS
ray-operator   Synced        Healthy
```